### PR TITLE
feat(node9&10): add `TextDecoder` and `TextEncoder` for `util` module

### PR DESCRIPTION
**Pull request by islishude**
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://nodejs.org/dist/latest-v10.x/docs/api/>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -6220,6 +6221,39 @@ declare module "util" {
         export function isWeakSet(object: any): object is WeakSet<any>;
         export function isWebAssemblyCompiledModule(object: any): boolean;
     }
+
+    // TypeScript Version: 2.1
```
Comment (Flarna) [inline types/node/index.d.ts]: What is the use of this comment?

File: types/node/v9/index.d.ts [hunk 2]
```diff
@@ -5659,6 +5660,38 @@ declare module "util" {
     export namespace promisify {
         const custom: symbol;
     }
+    // TypeScript Version: 2.1
```
Comment (Flarna) [inline types/node/v9/index.d.ts]: What is the use of this comment?

Comment (isLishude) [reply types/node/v9/index.d.ts]: I don't kown about it.if not has it,will get a error。

**Comment by isLishude** [comment]
Previous PR Travis error log

```
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/node/index.d.ts:6235:13
ERROR: 6235:13  expect  Compile error in typescript@2.0 but not in typescript@2.1.
Fix with a comment '// TypeScript Version: 2.1' just under the header.
Type expected.
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:101:19
    at Generator.next (<anonymous>)
    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:5:58)
    at <anonymous>
```

So I add it,and no any errror reports.

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
@isLishude I think the `// TypeScript Version: 2.1` line should be there just once just below the header, directly below `// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped` and not several times in the file. see e.g. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v8/index.d.ts#L26

**Comment by isLishude** [comment]
@Flarna I moved it on mine, but got the error, TypeScript version lower.

now I test ts version 2.2 and so on

```
Error: accepts depends on node but has a lower required TypeScript version.
    at checkTypeScriptVersions (/Users/lishude/DefinitelyTyped/node_modules/types-publisher/src/check-parse-results.ts:75:11)
    at Object.<anonymous> (/Users/lishude/DefinitelyTyped/node_modules/types-publisher/src/check-parse-results.ts:19:2)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/lishude/DefinitelyTyped/node_modules/types-publisher/bin/check-parse-results.js:4:58)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! definitely-typed@0.0.2 test: `node node_modules/types-publisher/bin/tester/test.js --run-from-definitely-typed`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the definitely-typed@0.0.2 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lishude/.npm/_logs/2018-05-14T12_28_55_740Z-debug.log
```

File: types/node/index.d.ts [hunk 3]
```diff
@@ -6220,6 +6221,39 @@ declare module "util" {
...
+        readonly encoding: string;
+        readonly fatal: boolean;
+        readonly ignoreBOM: boolean;
+        constructor(
+          encoding?: string,
+          options?: { fatal?: boolean; ignoreBOM?: boolean }
+        );
+        decode(
+          input?:
+            | Int8Array
```
Comment (Flarna) [inline types/node/index.d.ts]: The `| ` seems to be wrong here.

**Comment by Flarna** [comment]
I tried it locally and the initial error message that typescript 2.1 is need is simply a side effect of the extra `|` in `TextDecoder.decode`.
The followup error `Error: accepts depends on node but has a lower required TypeScript version.` is in the end correct as a module is not allowed require a higher typescript version as its dependents.
Seems this check works not correct if the `// TypeScript Version: 2.1` line is not part of the header.

So once the extra `| ` is removed the `// TypeScript Version: 2.1` lines can be also removed.

**Review by rbuckton — APPROVED** [review]

**Pull request merged**